### DISCUSSION TITLE
Time Delay

### DISCUSSION THREAD

**Discussion by mileslucas** [JuliaAstro/Transits.jl#18] [Ideas]
So `exoplanet` has an option for computing a light curve accounting for finite light-travel time. 
https://docs.exoplanet.codes/en/stable/tutorials/light-delay/
https://github.com/exoplanet-dev/exoplanet/blob/16cc4ef38258573e6a7f3fb4017c12f2fc19cbe7/src/exoplanet/orbits/keplerian.py#L417
https://github.com/rodluger/starry/issues/66

Is this the kind of thing that should be considered *de facto* for curves calculated with orbits? Should it be a keyword argument  for `compute(::LimbDark, ::Orbit, t, r)` (would that lead to performance drop?)? Should it be a wrapper (`TimeDelayLimbDark`)?

**Comment by icweaver** [comment]
I think that since the semi-major axis of pretty much all detected transiting exoplanets is well within half an AU, this effect isn't so pronounced. For hot Jupiters in particular (which make up the bulk of these detections), we don't really see a lag of more than ~30 seconds or so. I can definitely see a use case for multi-planet or more extended systems though, where high precision ephemerides probably play a more important role in making transit predictions. This might be currently out of scope though, what do you think?

**Reply by mileslucas** [reply]
Thanks for the "in the know" response. That makes me think that it would be more useful to have a wrapper type (either for light curves or for orbits) since this is more of a special case than de facto.
